src/psp.ts

```diff
@@ -0,0 +1,502 @@
+            );
+          }
+          // TODO: retryGo?
+          const resultDatas = await allocator.callStatic.multicall(staticCalldatas);
+          resultDatas.forEach((resultData: ethers.utils.BytesLike) => {
+            const result = allocator.interface.decodeFunctionResult('airnodeToSlotIndexToSlot', resultData);
+            if (result.subscriptionId && result.expirationTimestamp.gt(nowInSeconds())) {
+              subscriptionIds.push(result.subscriptionId);
+            }
+          });
+        }
+
+        // **************************************************************************
+        // Process each psp subscription in serial to keep nonces in order
+        // **************************************************************************
+        for (const subscriptionId of subscriptionIds) {
+          const subscriptionIdLogOptions = {
+            ...providerLogOptions,
+            additional: {
+              subscriptionId,
+            },
+          };
+
+          // **************************************************************************
+          // Fetch subscription details
+          // **************************************************************************
+          node.logger.debug('Fetching subscription details...', subscriptionIdLogOptions);
+
+          // On-chain details: AirnodeProtocol.subscriptions(id)
+          // Off-chain details: triggers.psp
+          const subscription = triggers.psp.find((s) => {
+            // TODO: Should the config contain subscriptionId field
+            // or should we derive it using the other fields to verify
+            // that values in config are correct?
+            return s.subscriptionId === subscriptionId;
+          });
+          if (isNil(subscription)) {
+            node.logger.warn('Subscription not found in triggers.psp', subscriptionIdLogOptions);
+            continue;
+          }
+
+          // **************************************************************************
+          // Check sponsorship status
+          // **************************************************************************
+          node.logger.debug('Checking sponsorship status...', subscriptionIdLogOptions);
+
+          if (!airnodeProtocol.sponsorToSubscriptionIdToPspSponsorshipStatus(subscription.sponsor, subscriptionId)) {
+            node.logger.info('Subscription not sponsored', subscriptionIdLogOptions);
+            continue;
+          }
+
+          // **************************************************************************
+          // Check authorization status
+          // **************************************************************************
+          node.logger.debug('Checking authorization status...', subscriptionIdLogOptions);
+
+          // TODO: This needs a little bit of thinking 🤔
+          // By default subscription.requester is set to the dapiServer address when calling dapiServer.registerBeaconUpdateSubscription()
```

Comment:
That's why checking the authorization doesn't make much sense, the Airnode operator probably wants to authorize the DapiServer contract